### ISSUE TITLE
Aedes chokes on numerical topic subpaths in retained messages

### ISSUE THREAD

**Issue by cefn** [mcollina/aedes#14]
I modified the https://github.com/mcollina/aedes/blob/master/benchmarks/pingpong.js benchmark as follows...

    client.publish('test', JSON.stringify(process.hrtime()), { qos: 1})

...changed to...

    client.publish(('/' + sent), JSON.stringify(process.hrtime()), { qos: 1, retain:true})

...and...

    client.subscribe('test', { qos: 1 }, publish)

...modified to...

    client.subscribe('#', { qos: 1 }, publish)

The server then reports like this...
```
server listening on port 1883 pid 84111
client error mqttjs_6aaa8387 no such packet
client error mqttjs_6aaa8387 no such packet
client error mqttjs_6aaa8387 no such packet
client error mqttjs_6aaa8387 no such packet
client error mqttjs_6aaa8387 no such packet
```

...and the client like this...

```
/usr/local/bin/node benchmarks/pingpong.js
sent/s 1286.8
sent/s 774.5999999999999
sent/s 795.6
offline
offline
offline
offline
```

By contrast if the same modified benchmark is run against Mosquitto, it reports...

```
/usr/local/bin/node benchmarks/pingpong.js
sent/s 2240
sent/s 1722.6
sent/s 1711.2
sent/s 1738.2
sent/s 1648.2
sent/s 1721.6
sent/s 1738.6
sent/s 1749.4
sent/s 1713.6
total 51751.08098199938
average 0.6259807550561179
mode [ 0.688611, 0.710682, 0.721568 ]
```

**Comment by mcollina** [comment]
In fact, it's not about numerical paths. It's about retained messages and QoS 1. Thanks for spotting this, I will get it fixed.

Beware that the in-memory storage for Aedes is definitely not optimized for storing 10k messages.

**Comment by mcollina** [comment]
This should be fixed by #17. Please test and reopen in case.
